extension/encoding/awscloudwatchmetricstreamsencodingextension/extension_test.go

```diff
@@ -17,7 +17,9 @@ func TestNew_JSON(t *testing.T) {
 	require.NotNil(t, e)
 
 	_, err = e.UnmarshalMetrics([]byte{})
-	assert.EqualError(t, err, errEmptyRecord.Error())
+	expectedErr := formatJSONError(errEmptyRecord)
+	require.Error(t, expectedErr)
+	assert.EqualError(t, err, expectedErr.Error())
```

Comment:
```suggestion
	assert.EqualError(t, err, `failed to unmarshal metrics as 'json' format: 0 metrics were extracted from the record`)
```

This way we can see what the error message is, and make sure it's meaningful to callers.